app/src/main/res/layout/dialog_critical_app_update.xml

```diff
@@ -0,0 +1,127 @@
+    android:layout_height="@dimen/spacing_1"
+    android:layout_marginTop="@dimen/spacing_24"
+    android:background="#DADDE0"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/updateNowButton" />
+
+  <com.google.android.material.textview.MaterialTextView
+    android:id="@+id/helpSectionTitle"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_24"
+    android:text="@string/critical_update_required_dialog_help_section_title"
+    android:textAppearance="?attr/textAppearanceBody0Medium"
+    android:textColor="?attr/colorOnSurface"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/divider" />
+
+  <com.google.android.material.textview.MaterialTextView
+    android:id="@+id/helpSectionDescription"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_16"
+    android:text="@string/critical_update_required_dialog_help_section_content"
+    android:textAppearance="?attr/textAppearanceBody0"
+    android:textColor="?attr/colorOnSurface"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/helpSectionTitle" />
+
+  <com.google.android.material.button.MaterialButton
+    android:id="@+id/shareOnWhatsappButton"
+    android:layout_width="@dimen/spacing_0"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_16"
+    android:elevation="@dimen/spacing_2"
+    android:text="@string/critical_update_required_dialog_help_section_phone_number"
+    android:textAppearance="?attr/textAppearanceButtonBig"
+    android:textColor="?attr/colorSecondary"
+    app:backgroundTint="?attr/colorSecondaryVariant"
+    app:elevation="@dimen/spacing_4"
+    app:icon="@drawable/ic_whatsapp_logo"
+    app:iconTint="?attr/colorSecondary"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/helpSectionDescription" />
+
+  <com.google.android.material.button.MaterialButton
+    android:id="@+id/shareOnTelegramButton"
+    android:layout_width="@dimen/spacing_0"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="@dimen/spacing_16"
+    android:elevation="@dimen/spacing_2"
+    android:text="@string/critical_update_required_dialog_help_section_phone_number"
+    android:textAppearance="?attr/textAppearanceButtonBig"
+    android:textColor="?attr/colorPrimary"
+    android:visibility="gone"
+    app:backgroundTint="?attr/colorPrimaryVariant"
+    app:elevation="@dimen/spacing_4"
+    app:icon="@drawable/ic_telegram_logo"
+    app:iconTint="@null"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@id/helpSectionDescription" />
```

Comment:
Instead of having 2 different buttons here, do you think we can instead have 2 styles and update them programmatically @janhavisinghh ? 🤔

Reply:
For example, how would the UI react when we update the json to have a different share type, how do we decide which one to show in this scenario? I was thinking of updating them programmatically, that way we can set a "default" button type that just renders the text. WDYT?